### PR TITLE
Account updated and deleted bytes towards quota

### PR DESCRIPTION
**Pull request by efirs**

### CODE REVIEW

File: server/services/v1/database/query_runner.go [hunk 1]
```diff
@@ -351,6 +355,8 @@ func (runner *DeleteQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 			}
 		}
 
+		request.IncrementWrittenBytes(ctx, len(row.Data.RawData))
```
Comment (ovaistariq) [inline server/services/v1/database/query_runner.go:358]: If I delete a document of size 2KB, will this cause it to be counted as two write units?

File: server/services/v1/database/query_runner.go [hunk 2]
```diff
@@ -247,7 +248,8 @@ func (runner *UpdateQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 
 		newData := internal.NewTableDataWithTS(row.Data.CreatedAt, ts, merged)
 		newData.SetVersion(coll.GetVersion())
-		// as we have merged the data, it is safe to call replace
+
+		request.IncrementWrittenBytes(ctx, len(merged))
```
Comment (himank) [inline server/services/v1/database/query_runner.go:252]: We need to update it for read queries that are scanning a lot of documents.

Comment (ovaistariq) [reply server/services/v1/database/query_runner.go:252]: Yes, we will need to do two things here for the Update:
- count the number of bytes read to fetch the data to be updated (read units consumed)
- count the amount of data written back (write units consumed)

File: server/services/v1/database/query_runner.go (hunk 1 above)
Comment (ovaistariq) [reply server/services/v1/database/query_runner.go:358]: Yes, we will need to do two things here for the Delete:
- count the number of bytes read to fetch the data to be deleted if we need to read. If it is a blind delete, then there is nothing read, so zero bytes (read units consumed)
- count the number of documents deleted as 1 write each if we are doing individual deletes
- if it is a blind delete then it is just 1 operation

**Comment by ovaistariq** [comment]
Is this ready? Could you also document the change in the PR description.

**Comment by efirs** [comment]
This is ready for update and delete, implemented on query_runner level. It's missing read part though.

@pboros suggested alternative approach using lower level, metrics KV wrapper. He is working on read part of it. If that approach is easier then we'll adopt it instead of this diff for the update and deletes too.

**Comment by ovaistariq** [comment]
Sounds good

**Comment by himank** [comment]
Closing as we don't need it.

**Pull request closed without merging**